turbopack/crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -521,29 +521,29 @@ impl FileSystem for DiskFileSystem {
         match &*self.read_dir_internal(fs_path).await? {
             InternalDirectoryContent::NotFound => Ok(DirectoryContent::not_found()),
             InternalDirectoryContent::Entries(entries) => {
-                let fs = fs_path.await?.fs;
-                let entries = entries
-                    .iter()
-                    .map(|(name, entry)| {
-                        let entry = match entry {
-                            InternalDirectoryEntry::File(path) => DirectoryEntry::File(
-                                FileSystemPath::new_normalized(fs, path.clone()),
-                            ),
-                            InternalDirectoryEntry::Directory(path) => DirectoryEntry::Directory(
-                                FileSystemPath::new_normalized(fs, path.clone()),
-                            ),
-                            InternalDirectoryEntry::Symlink(path) => DirectoryEntry::Symlink(
-                                FileSystemPath::new_normalized(fs, path.clone()),
-                            ),
-                            InternalDirectoryEntry::Other(path) => DirectoryEntry::Other(
-                                FileSystemPath::new_normalized(fs, path.clone()),
-                            ),
-                            InternalDirectoryEntry::Error => DirectoryEntry::Error,
-                        };
-                        (name.clone(), entry)
-                    })
-                    .collect();
-                Ok(DirectoryContent::new(entries))
+                let fs = *fs_path.await?.fs;
+                let normalize =
+                    |path: &RcStr| FileSystemPath::new_normalized(fs, path.clone()).to_resolved();
```

Comment:
The `normalize` closure here isn't something I'd expect a codemod to do, I just decided to clean this up a bit while I was here.

I'd expect a codemod to just put the `.to_resolved().await?` inline.

However, this is another example of a case where I had to eliminate `Iterator::map` and `Iterator::collect` calls.